src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1435,19 +1435,22 @@ private CSharpSyntaxNode GetBindingRoot(CSharpSyntaxNode node)
             }
 #endif
 
-            for (CSharpSyntaxNode current = node; current != this.Root; current = current.ParentOrStructuredTriviaParent)
+            if (EnableNullableAnalysis)
             {
-                if (current is StatementSyntax)
-                {
-                    return current;
-                }
+                // PROTOTYPE(nullable-api): This doesn't work for speculative models: the root is just the statement
+                // or expression being speculated on.
+                return Root;
```

Comment:
>return Root; [](start = 16, length = 12)

It would be good to add a comment explaining why is this the right thing to return the ```Root``` in this case.